packages/core/src/browser/keybinding.ts

```diff
@@ -603,14 +591,15 @@ export class KeybindingRegistry {
         this.keyboardLayoutService.validateKeyCode(keyCode);
         this.keySequence.push(keyCode);
         const bindings = this.getKeybindingsForKeySequence(this.keySequence);
-
-        if (bindings.partial.length === 0 && this.tryKeybindingExecution(bindings.full, event)) {
+        const full = bindings.full.find(binding => this.isEnabled(binding, event));
+        const partial = bindings.partial.find(binding => this.isEnabled(binding, event));
```

Comment:
I think `full.scope >= partial.scope` should be reversed an put in `find` that we check all enabled partial bindings not only first, i.e.
`const partial = bindings.partial.find(binding => (!full || binding.scope! > full.scope!) && this.isEnabled(binding, event));`

`scope!` because `getKeybindingsForKeySequence` cannot return a binding with `undefined` scope 

after that we can simply if-else branching also:
```ts
if (partial) {
    /* Accumulate the keysequence */
    event.preventDefault();
    event.stopPropagation();

    this.statusBar.setElement('keybinding-status', {
        text: `(${this.acceleratorForSequence(this.keySequence, '+')}) was pressed, waiting for more keys`,
        alignment: StatusBarAlignment.LEFT,
        priority: 2
    });
} else {
    if (full) {
        this.executeKeyBinding(full, event);
    }
    this.keySequence = [];
    this.statusBar.removeElement('keybinding-status');
}
```

Reply:
done